internal/uploader/auth_utils.go

```diff
@@ -1,73 +1,128 @@
 package uploader
 
 import (
+	"errors"
+	"fmt"
 	"strings"
 
 	"github.com/openshift/assisted-service/internal/cluster/validations"
 	"github.com/openshift/assisted-service/pkg/k8sclient"
-	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
 const (
-	openshiftTokenKey = "cloud.openshift.com"
+	openshiftNamespace  = "openshift-config"
+	openshiftSecretName = "pull-secret"
+	openshiftTokenKey   = "cloud.openshift.com"
 )
 
-func getPullSecret(clusterPullSecret string, k8sclient k8sclient.K8SClient, key string) (*validations.PullSecretCreds, error) {
-	data, exists := getOCMPullSecret(k8sclient)
-	if !exists {
-		data = clusterPullSecret
+type PullSecretCreds struct {
+	ManagementCreds *validations.PullSecretCreds
+	WorkloadCreds   *validations.PullSecretCreds
```

Comment:
Alternatively, 
```suggestion
type Credentials struct {
	Username string
	Email string 
	Upload   *validations.PullSecretCreds
```
So you wouldn't have to call `selectCredentials` [here](https://github.com/openshift/assisted-service/pull/6435/files#r1638404061) and can use each field individually

Reply:
Good idea, maybe `Identity { Email, Username }` and `AuthToken` or something?

Reply:
I pushed something, let me know if it fits what you had in mind

Reply:
Looks great to me, thanks!